### DISCUSSION TITLE
Is there a maintained setup for parsing the Rust language itself?

### DISCUSSION THREAD

**Discussion by tgross35** [pest-parser/pest#811] [Q&A]
Hey all,

Are there prewritten grammar files for established languages, specifically Rust? I'm hoping to be able to lex `.rs` files in a lightweight way. But if something exists already, it would be quite a nice time saver.

An index of off the shelf grammers for maintained languages would be pretty cool. Maybe this exists, but I have been unsuccessful finding it.

**Comment by tomtau** [comment]
I'm not aware of the Rust grammar in Pest, but there's this famous crate that uses the Rust compiler's internal parser: https://crates.io/crates/syn

**Reply by tgross35** [reply]
Lol, my end project was actually a tool to make designing `syn` parsing a little bit easier. But I don't believe `syn` does string -> `TokenTree` parsing, instead it does `TokenTree` -> tokens. `quote` does, but it relies on the compiler to parse - which isn't suitable for my case (hopefully targeting `wasm`).
